examples/cma_es_cartpole.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+"""
+This is an example to train a task with CMA-ES
+
+Here it runs CartPole-v1 environment with 100 epoches.
+
+Results:
+    AverageReturn: 100
+    RiseTime: epoch 38 (itr 760),
+              but regression is observed in the course of training.
+"""
+from garage.algos import CMAES
+from garage.baselines import LinearFeatureBaseline
+from garage.experiment import LocalRunner, run_experiment
+from garage.tf.envs import TfEnv
+from garage.tf.policies import CategoricalMLPPolicy
+from garage.tf.samplers import OnPolicyVectorizedSampler
+
+
+def run_task(*_):
+    """Train CMA_ES with Cartpole-v1 environment."""
+    with LocalRunner() as runner:
+        env = TfEnv(env_name="CartPole-v1")
+
+        policy = CategoricalMLPPolicy(
+            name="policy", env_spec=env.spec, hidden_sizes=(32, 32))
+
+        baseline = LinearFeatureBaseline(env_spec=env.spec)
+
+        runner.initialize_tf_vars()
+
+        n_samples = 20
+
+        algo = CMAES(
+            env_spec=env.spec,
+            policy=policy,
+            baseline=baseline,
+            max_path_length=100,
+            n_samples=n_samples)
+
+        runner.setup(algo, env, sampler_cls=OnPolicyVectorizedSampler)
+        # NOTE: make sure that n_epoch_cycles == n_samples !
```

Comment:
seems like `n = 2` for the instances which need a special runner constraint. how can we write this down in code instead of comments?

Reply:
Yes. And I want to elaborate on this constraint a little bit.

The `n_epoch_cycles` is defined as `number of calls to algo.train_once() in one epoch`. It controls how large an epoch is (in terms of iterations). It only concerns to **epoch-wise** logging and snapshotting, but not training (at least so far).  

So if someone breaks `n_epoch_cycles == n_samples` constraint, he/she will observe
- Unexpected snapshot frequency.
- Unexpected log of epoch time.

He/she will not observe
- Performance regression. (not learning, learn slowly, etc.)
- Wrong iteration-wise logging. (e.g. AverageReturn logged by sampler)

Reply:
And I am not in favor of subclassing `LocalRunner` for `CEM`  because I fear that once we add one algo-specific runner into the codebase, we end up adding many more sub-runner for new algos. In fact, off-policy algorithms also have this constraint. 

The conflict here is that our encapsulation is leaky.  Today we have constraint on `n_epoch_cycles` by algorithm, some other day in the future we might find a new constaint on some sampler parameters by algorithm. Decoupling is good, but also hard, [explained](https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/).

Reply:
alright sgtm